src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -57,22 +58,57 @@ private void captureOs() {
         sysProperty("os.name").ifPresent(buildScan::tag);
     }
 
+    private enum IdeaFlavour {
+        INTELLIJ("JetBrains", "IntelliJ IDEA"),
+        ANDROID_STUDIO("Google", "Android Studio");
+
+        private final String vendor;
+        private final String label;
+
+        IdeaFlavour(String vendor, String label) {
+            this.vendor = vendor;
+            this.label = label;
+        }
+
+        static Optional<IdeaFlavour> getByVendor(String str) {
+            return EnumSet.allOf(IdeaFlavour.class).stream()
+                    .filter(ide -> ide.vendor.equalsIgnoreCase(str))
+                    .findFirst();
+        }
+    }
+
     private void captureIde() {
         if (!isCi()) {
             // Wait for projects to load to ensure Gradle project properties are initialized
             gradle.projectsEvaluated(g -> {
+                Optional<String> ideaVendorName = sysProperty("idea.vendor.name");
+                Optional<String> ideaVersion = sysProperty("idea.version");
                 Optional<String> invokedFromAndroidStudio = projectProperty("android.injected.invoked.from.ide");
                 Optional<String> androidStudioVersion = projectProperty("android.injected.studio.version");
-                Optional<String> ideaVersion = sysProperty("idea.version");
                 Optional<String> eclipseVersion = sysProperty("eclipse.buildId");
                 Optional<String> ideaSync = sysProperty("idea.sync.active");
                 if (ideaSync.isPresent()) {
                     buildScan.tag("IDE sync");
                 }
-                if (invokedFromAndroidStudio.isPresent()) {
+                if (ideaVendorName.isPresent()) {
+                    IdeaFlavour.getByVendor(ideaVendorName.get()).ifPresent(ideaFlavour -> {
```

Comment:
can we make this simpler by comparing the strings instead of using an enum?

```
if (ideaVendorName.equalsIgnoreCase("Google")) {
  // tag with Android studio
  else (ideaVendorName.equalsIgnoreCase("Jetbrains") {
  // tag with Intellij
  }
  ```

Reply:
Let me adjust the current implementation that way

Reply:
You can now review the adjusted implementation @runningcode